### PR TITLE
test-case: Add multipipeline Pause-Resume Mediaplayback & Analog/i2s

### PR DESCRIPTION
**Pull request by stripes416**
simultaneous multiplayback on MediaPlayback and Analog/i2s pipelines

Signed-off-by: Kelly Ledford <kelly.ledford@intel.com>

### CODE REVIEW

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+if [ "$pcm_MPb" = "" ]||[ "$pcm_ANALOG" = "" ];
+then
+    dloge "No playback pcms are specified. Skip the pause-resume test"
+    exit 2
+fi
+
+[[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
+
+channel=2
+rate=4800
```
Comment (fredoh9) [inline test-case/check-pause-resume-multi-mediaplayback-analog.sh]:
```suggestion
rate=48000
```

Comment (stripes416) [reply test-case/check-pause-resume-multi-mediaplayback-analog.sh]: yep --- one too few 0s in that number. whoops.

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        send " "
+        after \$sleep_t
+        incr i
+        if { \$i > $repeat_count } { exit 0 }
+        exp_continue
+    }
+}
+exit 1
+MEDIAPLAYBACK
+expect <<ANALOG
```
Comment (xiulipan) [inline test-case/check-pause-resume-multi-mediaplayback-analog.sh]: will the two pause/resume run at the same time?
It looks like the second will start after the first finish.
@Bin-QA Could you help to check this patch

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        puts "\r(\$i/$repeat_count) MPb: Wait for \$sleep_t ms before resume"
+        send " "
+        after \$sleep_t
+        incr i
+        if { \$i > $repeat_count } { exit 0 }
+        exp_continue
+    }
+}
+exit 1
+MEDIAPLAYBACK
```
Comment (Bin-QA) [inline test-case/check-pause-resume-multi-mediaplayback-analog.sh]: please add `pid_of_playback=$!` to catch expect pid

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        incr i
+        if { \$i > $repeat_count } { exit 0 }
+        exp_continue
+    }
+}
+exit 1
+ANALOG
+
+ret=$?
+#flush the output
```
Comment (Bin-QA) [inline test-case/check-pause-resume-multi-mediaplayback-analog.sh]: please add check of `$ret`

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        send " "
+        after \$sleep_t
+        incr i
+        if { \$i > $repeat_count } { exit 0 }
+        exp_continue
+    }
+}
+exit 1
+ANALOG
+
```
Comment (Bin-QA) [inline test-case/check-pause-resume-multi-mediaplayback-analog.sh]: now you can  check expect of play exit status
```
wait pid_of_aplay
[[ $? -ne 0 ]] && dloge 
```

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+##    no errors occur for either process
+##
+
+source $(dirname ${BASH_SOURCE[0]})/../case-lib/lib.sh
+
+OPT_OPT_lst['M']='pcm_MPb'      OPT_DESC_lst['M']='MediaPlayback pcm for playback. Example: hw:0,0'
+OPT_PARM_lst['M']=1             OPT_VALUE_lst['M']=''
+
+OPT_OPT_lst['A']='pcm_analog'   OPT_DESC_lst['A']='Analog pcm for playback. Example: hw:1,0'
+OPT_PARM_lst['A']=1             OPT_VALUE_lst['A']=''
```
Comment (Bin-QA) [inline test-case/check-pause-resume-multi-mediaplayback-analog.sh]: How about use TPLG to go though this value?

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh (hunk 2 above)
Comment (Bin-QA) [reply test-case/check-pause-resume-multi-mediaplayback-analog.sh]: this design is good, run aplay refer expect script on the background is good idea
But I always think when the time increase, the random value will hard to sync time

Comment (stripes416) [reply test-case/check-pause-resume-multi-mediaplayback-analog.sh]: @xiulipan i did extensive testing on this -- they run at the same time. The point of this is that they are checking pause/resume of each pipeline independently. If they don't finish in-sync --- thats completely fine.

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh (hunk 3 above)
Comment (stripes416) [reply test-case/check-pause-resume-multi-mediaplayback-analog.sh]: will do.

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh (hunk 4 above)
Comment (stripes416) [reply test-case/check-pause-resume-multi-mediaplayback-analog.sh]: will do.

File: test-case/check-pause-resume-multi-mediaplayback-analog.sh (hunk 6 above)
Comment (stripes416) [reply test-case/check-pause-resume-multi-mediaplayback-analog.sh]: I'm not sure I understand what you mean here? Are you saying pull this value from the tplg file?

**Comment by stripes416** [comment]
closing --- due to #76

**Pull request closed without merging**